variables.tf

```diff
@@ -176,11 +180,23 @@ variable "node_pools" {
 }
 
 variable "node_pool_overrides" {
+  description = "Param override for var.node_pools"
+  type = map(
+    object({
```

Comment:
I don't believe this is correct. This limits the overrides to just the 3 pools.
The node_pools can have any number of pools of which 3 have default values.

Reply:
Should we update this to account for the x number of pools?
```
initial_node_count = local.node_pools.platform.initial_count + local.node_pools.compute.initial_count + local.node_pools.gpu.initial_count
```

Reply:
We shouldn't have to. It is there to handle https://github.com/hashicorp/terraform-provider-google/issues/6901. We have set remove_default_pool = true which causes them to be destroyed anyhow. It is really there to size the control plane which is now multi-AZ. Prior it was single AZ and a scale up caused an outage.